src/gui/shell.h

```diff
@@ -61,7 +61,7 @@ class Shell: public ShellWidget
 	void neovimMaximized(bool);
 	void neovimSuspend();
 	void neovimFullScreen(bool);
-	void neovimGuiCloseRequest();
```

Comment:
like wise on the C++ side make the exit code an optional argument with default 0.

Reply:
I fixed the mentioned issue by passing 0 to `neovimGuiCloseRequest` if there are no argument with the Close event.  Is this still needed?  (I can easily add this but will it be used from somewhere...?)

Reply:
I read this comment as changing the function signature to have a default parameter with value 0:
```C++
void neovimGuiCloseRequest(int status = 0);
```

No strong opinion, just my interpretation :)

Reply:
I also interpreted as you said, but I thought there would be no place calling `neovimGuiCloseRequest` with no argument.

Just curious.  I'm OK if it's about readability --- it's good to indicate that an argument of Close event is optional (I fixed it to have a default value 0).

Reply:
I noticed I've added the default argument to the wrong place (`mainwindow.h` which defines a slot connected to this signal).  Should I add it to both or signal only?

Reply:
It's just a convention thing... There will be different opinions depending on who you ask. I don't have a strong opinion here. Either way is fine with me.

In this case, a default parameter allows you to make this change and keep the old calling pattern valid. This might be useful if you want to exit Neovim, but don't care about status code (default to OK).

Reply:
Just for clarification... Sorry if you already know this!

Technically "optional" isn't the correct term for this language feature.

C++ supports "default arguments", which are different than an optional/keyword argument in some languages like Python. Default arguments simply have a default value when one is not provided.

For example, a call to `neovimGuiCloseRequest()` is converted by the compiler to `neovimGuiCloseRequest(0)` at compile-time.

Reply:
Thank you for your clarification!  Now it's much clearer for me.  I removed a default value from the slot's argument, and added one to the signal's argument.

You are true, "optional" was my bad wording.  I know it's just a compile time thing.   Sorry if I misunderstood as I'm new to Qt, but in my understanding basically signals and slots are used by connecting them each other (`connect(source, SIGNAL(neovimGuiCloseRequest(int)), receiver, SLOT(neovimGuiCloseRequest(int)))`).  From that point, the signature `neovimGuiCloseRequest(int)` is the same regardless of the default value's presence (as it's not a optional argument but just a default value), so at first I got confused.  However if there's default value in the signal's definition, it's certainly useful in the situation you described. I understand it's better to add default value to the signal's argument.

I'm sorry to take up your time.  Thank you for your detailed explanation!

Reply:
I use the word "optional" for this C++ feature too sometimes... No worries.

I think having the default parameter on the signal is the correct solution.

No problem, I'm glad to hear you're learning something new! :)